commit d0d87fec5abd5074a47c00a06bccf51b83a15f21
Author: Jan Kara <jack@suse.cz>
Date:   Wed Oct 20 22:32:02 2010 +0200

    udf: Call udf_add_free_space() for more blocks at once in udf_free_blocks()
    
    There's no need to call udf_add_free_space() for one block at a time. It saves
    us noticeable amount of work and yields different result from the original
    code only if the filesystem is corrupted and bitmap bit is already cleared.
    In such case counter of free blocks is probably wrong anyways so the change
    does not matter.
    
    Signed-off-by: Jan Kara <jack@suse.cz>

--- a/fs/udf/balloc.c
+++ b/fs/udf/balloc.c
@@ -157,10 +157,9 @@ static void udf_bitmap_free_blocks(struc
 				udf_debug("bit %ld already set\n", bit + i);
 				udf_debug("byte=%2x\n",
 					((char *)bh->b_data)[(bit + i) >> 3]);
-			} else {
-				udf_add_free_space(sb, sbi->s_partition, 1);
 			}
 		}
+		udf_add_free_space(sb, sbi->s_partition, count);
 		mark_buffer_dirty(bh);
 		if (overflow) {
 			block += count;
